certbot/tests/util.py

```diff
@@ -360,57 +360,63 @@ def setUp(self):
         self.config.chain_path = constants.CLI_DEFAULTS['auth_chain_path']
         self.config.server = "https://example.com"
 
+    """Acquire a file lock at lock_path and wait to release it.
+              # This will trigger the lock release in subprocess.
+              os.remove(trigger)
+      :param multiprocessing.Condition cv: condition for synchronization
+              process.communicate()
+      :param str lock_path: path to the file lock
+          assert process.returncode == 0
+      finally:
+          shutil.rmtree(workspace)
+
+
+      """
+
+
+def _handle_lock(event_in, event_out, path):
+    """
+    Acquire a file lock on given path, then wait for release it. This worker is coordinated
+    using events to signal the caller about the locking, and know when to release it.
+    :param multiprocessing.Event event_in: event object to signal when to release the lock
+    :param multiprocessing.Event event_out: event object to signal when the lock is acquired
+    :param path: the path to lock
+    """
+    if os.path.isdir(path):
+        my_lock = lock.lock_dir(path)
+    else:
+        my_lock = lock.LockFile(path)
+    try:
+        event_out.set()
+        event_in.wait(timeout=20)
```

Comment:
I think we have to check the return value here. See https://docs.python.org/3.7/library/threading.html#threading.Event.wait.

I think this can be as simple as adding an `assert`.

Reply:
Only available since 3.1, before that it returns nothing. It is true that if the timeout is reached, there is no exception. But the callback is likely to fail in this case. 

If you think it is a problem, we should use queues again. They throw exception when timeout is reached.

Reply:
It was [added in 2.7](https://docs.python.org/2.7/library/threading.html#threading.Event.wait) and 3.1. Since we only support 2.7 and 3.4+ this seems OK to me?

I do not think we should allow tests to potentially continue with the lock not properly obtained/released.

Reply:
I did not saw it for 2.7. In this case, yes it is OK. I implement that.

Reply:
Fixed.